crypto/bn/bn_lib.c

```diff
@@ -882,6 +882,84 @@ void BN_consttime_swap(BN_ULONG condition, BIGNUM *a, BIGNUM *b, int nwords)
 
 #undef BN_CONSTTIME_SWAP_FLAGS
 
+/*-
+ * Constant-time randomized conditional swap of a and b.
+ * a and b are swapped if condition is not 0.
+ * The randomization avoids creating a systematic bias in
+ * the values used by XOR operations, where the swap with condition==0
+ * does a lot of XORs that have one operand ('t' in the code below) at zero,
+ * while condition==1 results in 't' having about 50-50 zeroes and ones.
+ * This bias makes the swap vulnerable to EM (and likely other) physical
+ * side-channel attacks like Nonce@Once
+ * (for details, see: "Nonce@Once: A Single-Trace EM Side Channel Attack on
+ *  Several Constant-Time Elliptic Curve Implementations in Mobile Platforms"
+ * by M. Alam, B. Yilmaz, F. Werner, N. Samwel, A. Zajic, D. Genkin, Y. Yarom,
+ * and M. Prvulovic, in IEEE Euro S&P 2021)
+ *
+ * r is the value to use for randomization during the swap
```

Comment:
You talk about `r` here, but the function below seems to have `rand1` and `rand2`